### DISCUSSION TITLE
Color text in Console window?

### DISCUSSION THREAD

**Discussion by fcwze** [NationalSecurityAgency/ghidra#2849] [Q&A] [answered]
Is it possible to apply color formatting to text displayed in the console window by a user script? 

(I'm using Ghidra on Windows)

Thx

**Comment by dragonmacher** [comment, marked as answer]
~~There is no way to do this at this time.~~

Support for this was added to GhidraScript in 2025:
```
public void println(String message, Color color)
```

**Reply by astrelsky** [reply]
> There is no way to do this at this time.

There may not be an implementation in the api to do it but it should be possible since `printerr` prints in red right?

**Reply by dragonmacher** [reply]
> > There is no way to do this at this time.
> 
> There may not be an implementation in the api to do it but it should be possible since `printerr` prints in red right?

The console is using a JTextPane, the colors for which can be controlled via text attributes.   It is relatively easy to do.    Typically we would allow some color changing through options for a widget like this.   I cannot see exactly what options we could provide in this case, as I imagine users would like to color items in the console specific to what they are outputting.   Even from an API perspective, it is not clear how we could allow users to decorate text in a flexible manner.   

To answer these questions, we would need to know more about what users are trying to do.   If it is just changing the current colors, that would be easy to do via options.

**Reply by ryanmkurtz** [reply]
I picture this as an extension to the scripting API, like:

`println("Fail", Color.RAZZMATAZZ);`

**Reply by dragonmacher** [reply]
> I picture this as an extension to the scripting API, like:
> 
> `println("Fail", Color.RAZZMATAZZ);`

I'm pretty sure that `Color.RAZZMATAZZ` has been deprecated

**Reply by astrelsky** [reply]
I'm not familiar with `JTextPane` but I was thinking of using something like hex/html color codes to designate the color. It would be used in a similar fashion as how `Color.RAZZMATAZZ` is used above. `print("text", "#e3256b")` or `print("text", 0xe3256b)`
